### PR TITLE
Move to dotnet pack

### PR DESCRIPTION
**Pull request by jaredpar**
Changes our NuGet package creation process for our core packages to use `dotnet pack` instead of `nuget pack`. This gets us one step closer to being able to produce NuGet packages on Linux which we need in order to meet our source build requirements.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure

**Comment by jaredpar** [comment]
CC @333fred, @dseefeld

File: src/NuGet/NugetProjectPackUtil.csproj [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (333fred) [inline src/NuGet/NugetProjectPackUtil.csproj]: Rename for correct casing.

Comment (jaredpar) [reply src/NuGet/NugetProjectPackUtil.csproj]: What is the correct casing: nET?

Comment (333fred) [reply src/NuGet/NugetProjectPackUtil.csproj]: File name, not this particular line. I just can't make the comment on the file.

Comment (jaredpar) [reply src/NuGet/NugetProjectPackUtil.csproj]: Oh for FFS that didn't get fixed? I tried threed different times to fix the case on that file.

**Comment by jaredpar** [comment]
Did a VSTS official build to ensure I didn't break anything here https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1585390&_a=summary

**Comment by 333fred** [comment]
Differences in the signed build output:
* Some nupkgs are missing in the new output
	* DevDivPackages/ManagedDependencies
		* System.AppContext
		* System.Diagnostics.Process
		* System.IO.Compression
		* System.Net.Security
		* System.Reflection.TypeExtensions
		* System.Security.Cryptography.Algorithms
		* System.Security.Cryptography.X509Certificates
* Our version numbers in the test build are `-dev` instead of `-beta4`, and only `Prerelease` was built, not all 3 versions. @jaredpar says that this should be fixed by the latest commit.
* We added NuGet/NetFX20/Microsoft.NetFX20.<version>.nupkg with this change.

**Pull request merged**